### DISCUSSION TITLE
Simulation in the game - "Rules"

### DISCUSSION THREAD

**Discussion by astrolamb-gaming** [Thorium-Sim/thorium-nova#13] [General]
There are definite advantages and disadvantages to having lots of set "rules" for science fiction aspects of Nova. There is a pretty good book called "On Writing and Worldbuilding", which is based on a YouTube series (https://www.youtube.com/watch?v=YqC7IVKQPbg&list=PL1TLSKocOLTvdl54a6wgNdB8oSm-eZ7h6) which I think is applicable for this topic. The book focuses on magic systems, which honestly can include sci-fi, and touches a lot on the pros and cons of soft vs hard systems.

I think it is important to let the Flight Director (or mission builder, however that manifests itself) have the flexibility to use whatever systems they want to use for the story they are telling. We should have some built-in optional systems for use in our simulations, but the storyteller should be able to design their own or customize the built in systems to fit their storytelling needs. 

Alex mentioned in the meeting how Star Trek can be very inconsistent with their stories, which is something I absolutely agree with. But for our users, they should set their own constraints (or lack thereof). I don't think we should force any particular systems on them for how particular science fiction components (for lack of a better word) work (within the scope of a spaceship bridge sim of course).

**Comment by alexanderson1993** [comment]
This could be a bit of a challenge to get right. The issue is writing the necessary code for the universe simulation to handle every possible storytelling element that users want to write. By default there will always be some limitations on storytelling that must be addressed by us adding new features to the controls.

When it comes to systems, I assume you are talking to broader systems within the simulation which all play together. I would love some more examples of what you mean here. What constraints should storytellers be able to add or remove? What systems or science fiction components came to mind as you wrote this? What kind of flexibility might a storyteller want that doesn't come by default? Knowing the answers to these questions can help us figure out the best way to implement this without expecting storytellers to fork Thorium Nova and write the simulation code themselves.

The hope is that the Entity Component System framework that Thorium Nova uses will allow for some interesting, emergent interactions between the different simulations and entities, which might address this.

**Comment by astrolamb-gaming** [comment]
I don't recall what particular things I was thinking of when I wrote this last night. It may have largely been a gut reaction based on your comments about the inconsistencies star trek has in its "science". I think my main sticking point was that while star trek was often inconsistent, it annoys us because it should be all in the same universe, with the same rules for any particular circumstance applying to everyone. In a similar vein, Artemis SBS has a canonical story, with lots of things set fairly well in stone.

Thorium, on the other hand, doesn't need to have "canon" in its own right. Let the storytellers make their own canon. This will ineveitably be limited by what we have available for them, but if they want to build a new type of torpedo that creates a black hole when it detonates, let them.

I want to see the tools for Thorium to be extensive, and I agree that the ECS is a great asset in this goal. Having options to add components and triggers to any game object will allow that. In the case of the black hole torpedo, I envision that there would be a component that exclusively applies the black hole effect to any object its attached to. It could have various modifiers for range and strength, for example, and could then be added to a torpedo, or any other object the storyteller wants to use in that way.

**Reply by alexanderson1993** [reply]
Great example for the black hole torpedo. I think that's a reasonable expectation. 

Re. Thorium's canon, I 100% agree. I think there will be a "canon" that ships with Thorium (I wrote about this [last year on the blog](https://thoriumsim.com/blog/thorium-novas-lore)), and users are more than welcome to build on top of that, but I also agree that they should be able to start from scratch if they want to.

I think a good goal would be to make it possible to have stories set in the Thorium, Star Trek, Stargate, and Star Wars universes (and more? I'm open to other possibilities). This involves creating the necessary systems and configuration options for the simulation to operate the way the universe works in those universes, eg Warp Drive for Star Trek, Hyperdrive for Star Wars, wormhole rings for Stargate, etc.) Hard sci-fi should also be a possibility (even though hard sci-fi is boring 😛) .
